### PR TITLE
Added event data support to output modules WriteEvent method

### PR DESCRIPTION
**Pull request by joachimmetz**
Added event data support to output modules WriteEvent method

### CODE REVIEW

File: plaso/output/mediator.py [hunk 1]
```diff
@@ -89,72 +89,74 @@ def GetFormattedMessages(self, event):
...
     """
-    event_formatter = self.GetEventFormatter(event)
+    event_formatter = self.GetEventFormatter(event_data)
     if not event_formatter:
       return None
 
     return event_formatter.GetFormatStringAttributeNames()
 
-  def GetHostname(self, event, default_hostname='-'):
+  def GetHostname(self, event_data, default_hostname='-'):
```
Comment (Onager) [inline plaso/output/mediator.py]: Hostname is defined on event, not eventdata: https://github.com/log2timeline/plaso/blob/master/plaso/containers/events.py#L48 . See also comments on https://github.com/log2timeline/plaso/pull/2489

File: plaso/output/l2t_csv.py [hunk 2]
```diff
@@ -123,19 +156,12 @@ def _GetOutputValues(self, event):
...
-
-    hostname = self._FormatHostname(event)
-    username = self._FormatUsername(event)
+    inode = self._FormatInode(event_data)
+    hostname = self._FormatHostname(event_data)
+    username = self._FormatUsername(event_data)
 
     notes = []
-    note_string = getattr(event, 'notes', None)
+    note_string = getattr(event_data, 'notes', None)
```
Comment (Onager) [inline plaso/output/l2t_csv.py]: I don't think this is ever used, I don't see "notes" being populated on either the event or the event data.

File: plaso/formatters/asl.py [hunk 3]
```diff
@@ -47,26 +47,26 @@ class ASLFormatter(interface.ConditionalEventFormatter):
       7 : 'DEBUG'}
 
   # pylint: disable=unused-argument
-  def GetMessages(self, formatter_mediator, event):
-    """Determines the formatted message strings for an event object.
+  def GetMessages(self, formatter_mediator, event_data):
+    """Determines the formatted message strings for an event.
```
[the remark this answers is not in the document]
Comment (joachimmetz) [reply plaso/formatters/asl.py]: done

File: plaso/formatters/interface.py [hunk 4]
```diff
@@ -156,44 +156,46 @@ def GetFormatStringAttributeNames(self):
...
           EventLog resources.
-      event (EventObject): event.
+      event_data (EventData): event data.
 
     Returns:
       tuple(str, str): formatted message string and short message string.
 
     Raises:
-      WrongFormatter: if the event object cannot be formatted by the formatter.
+      WrongFormatter: if the event cannot be formatted by the formatter.
```
[the remark this answers is not in the document]
Comment (joachimmetz) [reply plaso/formatters/interface.py]: done

File: plaso/output/mediator.py (hunk 1 above)
Comment (joachimmetz) [reply plaso/output/mediator.py]: See #2489 for explanation. I've also created https://github.com/log2timeline/plaso/pull/2533

File: plaso/output/l2t_csv.py (hunk 2 above)
Comment (joachimmetz) [reply plaso/output/l2t_csv.py]: Out of scope of this PR but I've created https://github.com/log2timeline/plaso/issues/2534

**Pull request merged**